package/kedro_viz/data_access/repositories/modular_pipelines.py

```diff
@@ -210,6 +210,9 @@ def extract_from_node(self, node: GraphNode) -> Optional[str]:
 
         modular_pipeline = self.get_or_create_modular_pipeline(modular_pipeline_id)
 
+        # Inherit tags from the nodes of the modular pipeline.
+        modular_pipeline.inherit_tags(node.tags)

```

Comment:
This looks good. I see we inherit tags from node. The response in test_responses from `client.get("/api/main")` contains a modular pipeline as below. 
```json
 "uk": {
            "id": "uk",
            "name": "uk",
            "inputs": ["f1f1425b", "f0ebef01", "13399a82"],
            "outputs": ["d5a8b994"],
            "children": [
                {"id": "uk.data_science", "type": "modularPipeline"},
                {"id": "uk.data_processing", "type": "modularPipeline"},
            ],
        }
 ```
 
For this pipeline, the children are again modular pipelines. Should we consider inheriting the tags until we reach a node (like a depth first search ? ). This might be a reason why this node is not inheriting tags.

Reply:
Sounds reasonable. other than that looks fine to me.